lib/at_cmd_parser/at_cmd_decoder.c

```diff
@@ -0,0 +1,277 @@
+#include <zephyr.h>
+
+#include <at_cmd_parser/at_cmd_parser.h>
+#include <at_cmd_parser/at_cmd_decoder.h>
+
+#include <string.h>
+
+/* Forward declarations. */
+static void *at_CEREG_decode(struct at_param_list *param_list,
+			     u32_t valid_params);
+
+static void *at_CESQ_decode(struct at_param_list *param_list,
+			    u32_t valid_params);
+
+static void *at_NOT_CESQ_decode(struct at_param_list *param_list,
+				u32_t valid_params);
+
+static void *at_CMT_decode(struct at_param_list *param_list,
+			   u32_t valid_params);
+
+static void *at_CNUM_decode(struct at_param_list *param_list,
+			    u32_t valid_params);
+
+AT_CMD_DECODER_LIST_START(default_decoder_list)
+AT_CMD_DECODER_LIST_ENTRY("+CESQ", AT_CMD_CESQ_MODEL, at_CESQ_decode)
```

Comment:
It looks that this does not define a list of decoders, but list of tokens to parse. The naming is really confusing.

Reply:
Yes, I was not sure about the naming, I have tried to change AT_CMD_DECODER_LIST_ENTRY to AT_CMD_DECODER_LIST_ADD to see if that is any better. If you have better suggest please suggest.